File: HRM_systemAnguler/src/app/services/performancereview.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { PerformanceReview } from '../models/performancereview.model';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
@Injectable({
  providedIn: 'root',
})
export class PerformancereviewService {
  private apiUrl = `${environment.baseUrl}/api/performancereview`;
  constructor(private http: HttpClient) {}
  createPerformanceReview(review: PerformanceReview): Observable<PerformanceReview> {
    return this.http.post<PerformanceReview>(this.apiUrl, review);
  }
  getAllPerformanceReviews(): Observable<PerformanceReview[]> {
    return this.http.get<PerformanceReview[]>(this.apiUrl);
  }
  getPerformanceReviewById(id: number): Observable<PerformanceReview> {
    return this.http.get<PerformanceReview>(`${this.apiUrl}/${id}`);
  }
  updatePerformanceReview(id: number, review: PerformanceReview): Observable<PerformanceReview> {
    return this.http.put<PerformanceReview>(`${this.apiUrl}/${id}`, review);
  }
  deletePerformanceReview(id: number): Observable<string> {
    return this.http.delete(`${this.apiUrl}/${id}`, {
      responseType: 'text',
    });
  }
}

File: HRM_system/src/main/java/com/emranhss/HRM_system/performancereview/PerformanceReviewService.java
package com.emranhss.HRM_system.performancereview;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface PerformanceReviewService {

    
    PerformanceReviewResponseDto createPerformanceReview(
            PerformanceReviewRequestDto dto);

    
    List<PerformanceReviewResponseDto> getAllPerformanceReviews();

    
    PerformanceReviewResponseDto getPerformanceReviewById(Long id);

    
    PerformanceReviewResponseDto updatePerformanceReview(
            Long id,
            PerformanceReviewRequestDto dto);

    
    void deletePerformanceReview(Long id);


}
